core/api/src/app/accounts/get-account-transactions-for-contact.ts

```diff
@@ -8,29 +9,50 @@ import { WalletsRepository } from "@/services/mongoose"
 export const getAccountTransactionsForContact = async ({
   account,
   contactUsername,
-  paginationArgs,
+  rawPaginationArgs,
 }: {
   account: Account
   contactUsername: Username
-  paginationArgs?: PaginationArgs
-}): Promise<PaginatedArray<WalletTransaction> | ApplicationError> => {
+  rawPaginationArgs: {
+    first?: number
+    last?: number
+    before?: string
+    after?: string
+  }
+}): Promise<PaginatedQueryResult<WalletTransaction> | ApplicationError> => {
+  const paginationArgs = checkedToPaginatedQueryArgs({
+    paginationArgs: rawPaginationArgs,
+    maxPageSize: MAX_PAGINATION_PAGE_SIZE,
+  })
+
+  if (paginationArgs instanceof Error) return paginationArgs
+
   const ledger = LedgerService()
 
   const wallets = await WalletsRepository().listByAccountId(account.id)
   if (wallets instanceof Error) return wallets
 
-  const resp = await ledger.getTransactionsByWalletIdAndContactUsername({
+  const ledgerTxs = await ledger.getTransactionsByWalletIdAndContactUsername({
     walletIds: wallets.map((wallet) => wallet.id),
     contactUsername,
     paginationArgs,
   })
-  if (resp instanceof LedgerError) return resp
+  if (ledgerTxs instanceof LedgerError) return ledgerTxs
+
+  const nonEndUserWalletIds = Object.values(await getNonEndUserWalletIds())
+
+  const txEdges = ledgerTxs.edges.map((edge) => {
+    const { transactions } = WalletTransactionHistory.fromLedger({
+      ledgerTransactions: [edge.node],
+      nonEndUserWalletIds,
+      memoSharingConfig,
+    })
 
-  const confirmedHistory = WalletTransactionHistory.fromLedger({
-    ledgerTransactions: resp.slice,
-    nonEndUserWalletIds: Object.values(await getNonEndUserWalletIds()),
-    memoSharingConfig,
+    return {
+      cursor: edge.cursor,
+      node: transactions[0],
```

Comment:
is it possible that transactions has length 0?

Reply:
Actually I will just change the method to take in a single tx instead of a list.